examples/gno.land/r/gov/dao/v3/memberstore/types.gno

```diff
@@ -0,0 +1,172 @@
+package memberstore
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+)
+
+type ErrMemberAlreadyExists struct {
+	Tier string
+}
+
+func (e *ErrMemberAlreadyExists) Error() string {
+	return "member already exists on tier " + e.Tier
+}
+
+type Member struct {
+	InvitationPoints int
+}
+
+func (m *Member) RemoveInvitationPoint() {
+	if m.InvitationPoints <= 0 {
+		panic("not enough invitation points")
+	}
+
+	m.InvitationPoints = m.InvitationPoints - 1
+}
+
+// MembersByTier contains all `Member`s indexed by their Address.
+type MembersByTier struct {
+	*avl.Tree // tier name -> address -> member
+}
+
+func NewMembersByTier() MembersByTier {
+	return MembersByTier{avl.NewTree()}
+}
+
+// GetTierSize tries to get how many members are on the specified tier. If the tier does not exists, it returns 0.
+func (mbt MembersByTier) GetTierSize(tn string) int {
+	tv, ok := mbt.Get(tn)
+	if !ok {
+		return 0
+	}
+
+	tree, ok := tv.(*avl.Tree)
+	if !ok {
+		return 0
+	}
+
+	return tree.Size()
+}
+
+// SetMember adds a new member to the specified tier. The tier index is created on the fly if it does not exists.
+func (mbt MembersByTier) SetMember(tier string, addr std.Address, member *Member) error {
+	_, t := mbt.GetMember(addr)
+	if t != "" {
+		return &ErrMemberAlreadyExists{Tier: t}
+	}
+
+	if ok := mbt.Has(tier); !ok {
+		mbt.Set(tier, avl.NewTree())
+	}
+
+	ms, _ := mbt.Get(tier)
+	mst := ms.(*avl.Tree)
+
+	mst.Set(string(addr), member)
+
+	return nil
+}
+
+// GetMember iterate over all tiers to try to find a member by its address. The tier ID is also returned if the Member is found.
+func (mbt MembersByTier) GetMember(addr std.Address) (m *Member, t string) {
+	mbt.Iterate("", "", func(tn string, msv interface{}) bool {
+		mst, ok := msv.(*avl.Tree)
+		if !ok {
+			panic("MembersByTier values can only be avl.Tree")
+		}
+
+		return mst.Iterate("", "", func(ma string, msv interface{}) bool {
```

Comment:
We should replace this with a direct getter. You can use the address as the key for this or another AVL tree, or consider a composite key. However, avoid iteration in this case.

Reply:
hum, if i'm not wrong the key is already good, so it's just a matter of replace this iteration with a simple .Get, maybe.

Reply:
changed